test.py

```diff
@@ -414,7 +414,10 @@ def __init__(self, path, cfg: dict, options: argparse.Namespace, mode: str) -> N
 
         cluster_cfg = self.cfg.get("cluster", {"initial_size": 1})
         cluster_size = cluster_cfg["initial_size"]
-        pool_size = cfg.get("pool_size", 2)
+        if options.ci:
+            pool_size = cfg.get("ci_pool_size",  8)
+        else:
+            pool_size = cfg.get("pool_size", 2)
```

Comment:
I never understood where this constant "pool size" comes from.
It looks like something that should have been calculated somehow from the number of CPUs and amount of memory (and taking into account the number of CPUs and memor used by the individual tests), and not something that should be in the code base.

Reply:
Truly agree that pool size should not be hard-coded and should adapt to the machine. But it's not possible with the current state of the framework.

Reply:
@nyh , your suggestion also will work only if we have one common pool, in future we will move into this direction([doc](https://docs.google.com/document/d/1iKQbfqEYPp2KoaKSuMX7RMZ6BE-9ZAXwTVoYyYfUdW4/edit), milestone3).

but now we just try to speed ci using what test.py already has with minimal changes/efforts.